javascript/implementation/Database.ts

```diff
@@ -81,10 +73,30 @@ export class Database {
                 break;
             }
         }
-        if (!this.myChain) {
-            await this.startChain(identity);
+        if (reused) {
+            ensure(reused.medallion > 0);
+            this.myChain = reused;
+        } else {
+            const medallion = makeMedallion();
+            const chainStart = generateTimestamp();
+            const bundler = new Bundler(this.identity, medallion);
+            const bundleInfo = bundler.seal({
+                medallion, timestamp: chainStart, chainStart
+            });
+            ensure(bundleInfo.comment == this.identity);
+            const commitBytes = bundler.bytes;
+            await this.store.addBundle(commitBytes, true);
+            this.myChain = (await this.store.getClaimedChains()).get(medallion);
```

Comment:
I don't think this ought to be necessary.  You already know the chainStart and medallion for the chain that you started, so you shouldn't need to call getClaimedChains.

Reply:
Yeah it feels pretty clunky, but I don't have access to the actorId to build a ClaimedChain object.